public/js/main.js

```diff
@@ -15,5 +20,64 @@
                 $scope.loginUri = result.data.uri;
             });
         });
+
+        function displayConversation(newConversationData) {
+            var conversationIdx = $scope.conversations.findIndex(function(conversation) {
+                return conversation.data.id === newConversationData.id;
+            });
+            if (conversationIdx === -1) {
+                $scope.conversations.push({
+                    data: newConversationData,
+                    messageEntryText: "",
+                    messages: []
+                });
+                getMessages($scope.conversations.length - 1);
+            } else {
+                $scope.conversations[conversationIdx].data = newConversationData;
+                getMessages(conversationIdx);
+            }
+        }
+
+        function openConversation(recipientID) {
+            $http.get("/api/conversations/" + recipientID).then(function(conversationResult) {
+                displayConversation(conversationResult.data);
+            }, function() {
+                $http({
+                    method: "POST",
```

Comment:
The intention is that the client first attempts to GET the conversation and display it, and if that fails then it will instead POST a new conversation and display the newly created conversation, I'll try and clarify this in the code - although if you think the problem is more structural, then I'll try to fix the code instead